server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -197,7 +197,27 @@ class LivyServer extends Logging {
           livyConf.get(AUTH_KERBEROS_NAME_RULES))
         server.context.addFilter(holder, "/*", EnumSet.allOf(classOf[DispatcherType]))
         info(s"SPNEGO auth enabled (principal = $principal)")
-
+      case authType@LdapAuthenticationHandler.TYPE =>
+        val holder = new FilterHolder(new AuthenticationFilter())
+        holder.setInitParameter(AuthenticationFilter.AUTH_TYPE, authType)
+        val url = livyConf.get(LivyConf.AUTH_LDAP_URL)
+        if (url != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.PROVIDER_URL, url)
+        }
+        val baseDN = livyConf.get(LivyConf.AUTH_LDAP_BASE_DN)
+        if (baseDN != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.BASE_DN, baseDN)
+        }
+        val domain = livyConf.get(LivyConf.AUTH_LDAP_USERNAME_DOMAIN)
+        if (domain != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.LDAP_BIND_DOMAIN, domain)
+        }
```

Comment:
Either of them can be null but I can't put then in holder if any is null because it internally uses HashTable for env variables. It will throw NPE in this case. But I can simplify code a bit.